tests/python/unittest/test_higher_order_grad.py

```diff
@@ -129,6 +131,44 @@ def check_second_order_unary(x, op, grad_grad_op):
     # Validate the gradients.
     assert_almost_equal(expected_grad_grad, x.grad.asnumpy())
 
+class RandomShapes(object):
+    def __init__(self, dim):
+        self.dim = dim
+        self.curdim = 1
+
+    def __iter__(self):
+        return self
+
+    def next(self):
+        return self.__next__()
+
+    def __next__(self):
+        if self.curdim > self.dim:
+            raise StopIteration
+        shape = rand_shape_nd(self.curdim)
+        print(shape)
+        x = nd.random.normal(shape=shape)
+        self.curdim += 1
+        return x
+
+
+@with_seed()
+def test_dense_backward():
+    import mxnet.autograd as ag
+    import mxnet.ndarray as nd
+    for x in RandomShapes(5):
+        net = gluon.nn.Sequential()
+        with net.name_scope():
+            #net.add(gluon.nn.Dense(1, in_units=x.shape[1]))
+            net.add(gluon.nn.Dense(1))
+        net.initialize(mxnet.initializer.Constant(.5))
+        x.attach_grad()
+        with ag.record():
+            y = net.forward(x)
+            x_grad = ag.grad(y, x, create_graph=True, retain_graph=True)[0]
```

Comment:
We'd better do something like the following to check the correctness.
```python
        with ag.record():
            y = net.forward(x)
            x_grad = ag.grad(y, x, create_graph=True, retain_graph=True)[0]
            z = (random_multiplier * x_grad).sum()
        z.backward()
```
Then we check whether `x.grad` is correct.